File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem09_StringCompression.java
package org.phoenix.leetcode.challenges;

public class Problem09_StringCompression {
    public int compress(char[] chars) {
        int index = 0;
        int i = 0;
        while (i < chars.length) {
            int j = i;
            while (j < chars.length && chars[i] == chars[j]) {
                j++;
            }
            chars[index++] = chars[i];
            if (j - i > 1) {
                String count = String.valueOf(j - i);
                for (char c : count.toCharArray()) {
                    chars[index++] = c;
                }
            }
            i = j;
        }
        return index;
    }
}

File: JavaScript/leetcode75/src/main/Problem09_StringCompression.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function (chars) {
  let index = 0;
  let i = 0;
  while (i < chars.length) {
    let j = i;
    while (j < chars.length && chars[i] == chars[j]) {
      j++;
    }
    chars[index++] = chars[i];
    if (j - i > 1) {
      let count = j - i + "";
      for (let c of count) {
        chars[index++] = c;
      }
    }
    i = j;
  }
  return index;
};

module.exports = compress;
